### ISSUE TITLE
theme  don't apply styles to Input

### ISSUE THREAD

**Issue by musetraine** [GeekyAnts/NativeBase#3856]
theme  don't apply styles _focus: {} to Input, but if I put it in JSX it is work
`Input: {
			baseStyle: (props: Record<string, any>) => {
				return {
					placeholderTextColor: 'light.300',
					borderColor: 'gray.50',
					backgroundColor: 'rgba(255, 255, 255, 0.6)',
					_focus: {
						// style: {
						// 	borderColor: '#FF503E',
						// },
						borderColor: 'success.50',
					},
					_invalid: {
						borderColor: 'error.50'
					}
				}
			},
			defaultProps: {
				_focus: {
					borderColor: 'success.50',
				}
			},
		}`

**Comment by rayan1810** [comment]
@musetraine can you please share a [Expo Snack](snack.expo.dev) to reproduce this issue. I'm unable to reproduce this or maybe doing something different.

**Comment by cruzlutor** [comment]
I'm having a similar issue, spacing props are not working for extendTheme on the input component, here is my code

```
const theme = extendTheme({
  components: {
    Input: {
      sizes: {
        md: {
          py: 8,
          px: 5,
          bg: "#000000",
        },
      },
    },
  },
});

...

<FormControl isRequired mb={4}>
  <Input placeholder="Email" />
</FormControl>
<FormControl isRequired mb={4}>
  <Input placeholder="Password" />
</FormControl>
<Button mt={6}>
  Signin
</Button>
```

This is the result
![Screen Shot 2021-08-11 at 8 44 48 PM](https://user-images.githubusercontent.com/1368971/129125515-432c2953-6c5c-4a2a-a31c-0aeb5b7bb17e.png)

As you can see, vertical padding is not changing at all but the backgroundColor does

**Comment by cruzlutor** [comment]
Ok, finally I make it work, but I have to do it in this way

```
const theme = extendTheme({
  components: {
    Input: {
      sizes: {
        md: {
          _ios: {
            py: 4,
            px: 5,
          },
          _android: {
            py: 4,
            px: 5,
          },
          _web: {
            py: 4,
            px: 5,
          },
        },
      },
    },
  },
});
```

It looks a little ugly, I suppose generic props on `extendTheme` are not overwriting OS-specific props maybe?https://github.com/GeekyAnts/NativeBase/blob/development/src/theme/components/input.ts#L44

**Comment by stefanosandes** [comment]
> extendTheme does not apply borderColor styling to the _focus element when extending the Input object. All styles seem to be applied correctly for the _focus element, except for the borderColor property. Setting this inline seems to work, but the extending the default theme does not support the borderColor value specified in the _focus element.
> 
> ```
> export const BaseTheme = extendTheme({
>   components: {
>     Input: {
>       baseStyle: {
>         color: 'secondary.500',
>         borderColor: 'secondary.500',
>         _focus: {
>           color: 'red.900',
>           borderColor: 'red.900',
>           height: '150px',
>         },
>       },
>     },
>   }
> }
> ```

Same problem here =/

**Comment by matheusbaumgart** [comment]
Even after setting all variants I still can't change the `borderColor` of a focused Input.

**Comment by joseortiz9** [comment]
Still happening on last version `3.4.11`, can not change focus or hover from input theme styles, have to directly added to the component.

```js
<Input
  _focus={{ borderColor: 'text.900' }}
  {...props}
/>
```

I tried different ways, non of them work.
```
baseStyles: {
  _focus: {
    style: {
      borderColor: 'red'
    }
  }
}
```
```
baseStyles: {
  _focus: {
      borderColor: 'red'
  }
}
```
```
baseStyles: {
_light: {
  _focus: {
      borderColor: 'red'
  }
}
}
```
```
...many more
```

**Comment by icgarlin** [comment]
I have the same problem as @joseortiz9

**Comment by Viraj-10** [comment]
Hi, try Following code
```
 <Input
      focusOutlineColor="red.900"
      _light={{
        focusOutlineColor: 'red.900',
        _focus: {
          borderColor: 'red.900',
          _hover: { borderColor: 'red.900' },
        },
      }}
      placeholder="Enter your name"
    />
```
for now, we understand it's a difficult API we will try to fix it.

**Comment by ankit-tailor** [comment]
Thanks for your patience. This issue is fixed in `3.4.15`.
